editions/tw5.com/tiddlers/howtos/Concatenating text and variables using macro substitution.tid

```diff
@@ -0,0 +1,57 @@
+created: 20160424150551727
+modified: 20161223225338631
+tags: Learning
+title: Concatenating text and variables using macro substitution
+type: text/vnd.tiddlywiki
+
+It's a frequent use case in ~TiddlyWiki that you will want to put the results of variables together with various bits of strings of text. This process in some programming languages is often referred to as "concatenating" text.
+
+You might, for instance want to set up a template for your customer database, where links will automatically refer to additional contact information about your customer. Inside your tiddler, you might try something like this:
+
+```
+* [[Additional Info|<<currentTiddler>>-Contact]]
+```
+
+But that won't work. If you try this, the link will be interpreted very literally, and will attempt to take you to:
+
+```
+<<currentTiddler>>-Contact
+```
+
+The solution is to use a macro to put the rendered value of `<<currentTiddler>>` together with the bit of additional text, `-Contact`.
+
+Create a macro either at the top of the tiddler or in a global tiddler (see [[Macros]]) like this:
+
+```
+\define linkup(link) [[Additional Info|$link$-Contact]]
+```
+
+You might be tempted to invoke the new macro like this:
+
+```
+<<linkup <<currentTiddler>> >>
+```
+
+But if you do, you will find that `<<currentTiddler>>` doesn't get rendered, but instead gets passed literally.
+
+Instead, use the [[MacroCallWidget]] widget, like this:
+
+```
+<$macrocall $name="linkup" link=<<currentTiddler>> />
+```
+
+In this case, we passed the value of a variable directly to our macro. This is the most general way to go about this task. If you wanted to create more links based on other variables you could re-use the macro for each situation.
```

Comment:
This is a stylistic difference. Maybe wrapping macro calls with variables is more generic, but personally I think it's ugly when I have to wrap a macro call with variables. This step is unnecessary when calling functions in most real programming language. I'll think about adding the additional example.